BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -417,14 +417,14 @@ protected virtual HttpRequestMessage CreateRequestMessage(Uri address, Stream co
                 // TODO: NEED TO USE CACHING HERE
                 var authToken = this.CredentialEnvelope.GetToken();
 
-                if (authToken == null)
+                if (authToken == default(AuthToken))
                 {
                     // TODO: DO NOT SEND. RETURN FAILURE AND LET CHANNEL DECIDE WHEN TO RETRY.
                     // This could be either a configuration error or the AAD service is unavailable.
                 }
                 else
                 {
-                    request.Headers.TryAddWithoutValidation(AuthConstants.AuthorizationHeaderName, AuthConstants.AuthorizationTokenPrefix + authToken);
+                    request.Headers.TryAddWithoutValidation(AuthConstants.AuthorizationHeaderName, AuthConstants.AuthorizationTokenPrefix + authToken.Token);
```

Comment:
would we eventually be modifying this section, to look at authToken.ExpiresOn, and retrieve new Token, if needed?

Reply:
Not at the moment. 
The Azure.Identity library handles *most* lifecycle scenarios (caching, renewals, expiration).

There's ongoing conversations about how to handle scenarios not yet covered by Azure.Identity.